### PR TITLE
Adds createPropNameIDFromSymbol

### PR DESCRIPTION
**Pull request by TatianaKapos**
Paired PR to https://github.com/microsoft/hermes-windows/pull/94, 
TDLR: Adds in createPropNameIDFromSymbol to match facebook/hermes JSI

Updated all JSI files to match to https://github.com/facebook/hermes.git
Commit hash:	993a407ec80359e47d2bfac5b17ac533e58e2ed5
Author:		Anand Bashyam <kalyanavarathan@fb.com>#mailto:kalyanavarathan@fb.com
Date:		4/18/2022 10:08:22 AM

A new README.md file is added in the JSI folder about the matching Hermes commit.

Other related changes:

* Fixed HostObjectProxy to work with Symbol PropID and with the prototype chain.
* Formatted code in the changed files.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/v8-jsi/pull/121)

### CODE REVIEW

**Comment by vmoroz** [comment]
We have two implementations of JSI API in this repo. Please make sure that they both compile and all unit tests continue to run.
[https://github.com/microsoft/v8-jsi/blob/master/src/public/V8JsiRuntime.h](https://github.com/microsoft/v8-jsi/blob/master/src/public/V8JsiRuntime.h)
[https://github.com/microsoft/v8-jsi/blob/master/src/public/NapiJsiRuntime.h](https://github.com/microsoft/v8-jsi/blob/master/src/public/NapiJsiRuntime.h)

File: src/V8JsiRuntime.cpp [hunk 1]
```diff
@@ -452,7 +452,7 @@ std::once_flag g_tracingInitialized;
 
 void V8Runtime::initializeTracing() {
 #ifdef _WIN32
-  std::call_once(g_tracingInitialized, [](){ globalInitializeTracing(); });
+  std::call_once(g_tracingInitialized, []() { globalInitializeTracing(); });
```
Comment (TatianaKapos) [inline src/V8JsiRuntime.cpp:455]: Looks like clang-format picked up a lot of fixes

Comment (vmoroz) [reply src/V8JsiRuntime.cpp:455]: These are just a few lines. I do not think that it is an issue. Though, we usually prefer all such formatting changes going into a separate PR.

Comment (TatianaKapos) [reply src/V8JsiRuntime.cpp:455]: okay good to know! I'll leave it then and make sure to note for next time :)

File: src/public/NapiJsiRuntime.cpp [hunk 2]
```diff
@@ -626,6 +627,11 @@ facebook::jsi::PropNameID NapiJsiRuntime::createPropNameIDFromString(const faceb
   return MakePointer<facebook::jsi::PropNameID>(uniqueStr);
 }
 
+facebook::jsi::PropNameID NapiJsiRuntime::createPropNameIDFromSymbol(const facebook::jsi::Symbol &sym) {
+  CHECK_ELSE_CRASH(false, "Not implemented");
```
Comment (vmoroz) [inline src/public/NapiJsiRuntime.cpp:631]: > CHECK_ELSE_CRASH(false, "Not implemented"); [](http://example.com/codeflow?start=1&length=44)

Please note that I am deleting this file in
[https://github.com/microsoft/v8-jsi/pull/122](https://github.com/microsoft/v8-jsi/pull/122)

The real fix for this code must go to RNW implementation: 
[https://github.com/microsoft/react-native-windows/blob/main/vnext/Microsoft.ReactNative.Cxx/JSI/NodeApiJsiRuntime.cpp](https://github.com/microsoft/react-native-windows/blob/main/vnext/Microsoft.ReactNative.Cxx/JSI/NodeApiJsiRuntime.cpp)

Comment (vmoroz) [reply src/public/NapiJsiRuntime.cpp:631]: Please ignore this comment: based on the feedback I had restored the deleted files.
Let me know if I can help with implementing the change in the NapiJsiRuntime.cpp.

Comment (TatianaKapos) [reply src/public/NapiJsiRuntime.cpp:631]: okay awesome! I actually have a branch that fixes the changes in react-native-windows (https://github.com/TatianaKapos/react-native-windows/commit/43293c11e96f3832ceb861d4905886e18ef6ba84) and will make a PR once these gets merged in.

File: config.json [hunk 3]
```diff
@@ -1,4 +1,4 @@
 {
-    "version": "0.65.12",
```
[the remark this answers is not in the document]
Comment (vmoroz) [reply config.json:2]: Good point! Let me change it.

**Review by vmoroz — APPROVED** [review]

**Review by tudorms — APPROVED** [review]
:shipit:

**Pull request merged**